platforms/core-configuration/model-core/src/main/java/org/gradle/internal/reflect/DefaultTypeValidationContext.java

```diff
@@ -53,7 +53,9 @@ private DefaultTypeValidationContext(@Nullable Class<?> rootType, boolean report
         this.reportCacheabilityProblems = reportCacheabilityProblems;
     }
 
-    public static final DefaultProblemCategory MISSING_NORMALIZATION_CATEGORY = new DefaultProblemCategory(VALIDATION + SEPARATOR + MISSING_NORMALIZATION_ANNOTATION);
+    public static final String MISSING_NORMALIZATION_CATEGORY_DETAILS ="property" + SEPARATOR + "missing-normalization-annotation";
```

Comment:
❓ Separator is a "." in here. In addition, you use the fields verbatim. Should this also be kabab-case?
https://github.com/gradle/gradle/pull/26895/files#diff-e85f4ecdbd8f5058e2dde8eb8106c71571c3f8e1e28688b8546cb3bfad48ac90R122

Reply:
Correct me if I'm wrong, but the separator is `:`
https://github.com/gradle/gradle/blob/01d0b2c7919eb653980487bba73245b7b93dec8f/platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemCategory.java#L36

https://github.com/gradle/gradle/blob/c9ca663aee2e95c79250c8af3abfb2315e53a4a9/platforms/core-runtime/base-services/src/main/java/org/gradle/util/Path.java#L40